cdsp/information-layer/README.md

```diff
@@ -0,0 +1,233 @@
+- database handler: Interaction with the chosen database, configurable
+- router: API provider that connects to the database handler
+
+```mermaid
+flowchart LR
+    client <--> router
+    router <--> handlers
+    handlers <--> realm_handler
+    handlers <--> iotdb_handler
+    realm_handler <--> realmdb
+    iotdb_handler <--> iotdb
+```
+
+# Hello World Setup
+
+### Build the typescript application natively
+
+Make sure all libraries are installed
+```bash
+npm install
+```
+Build the application
+```bash
+npm run build
+```
+Run the unit tests
+```bash
+npx jest
+```
+or
+```bash
+npx jest --verbose
+```
+
+## Backend Option 1: IoTDB
+
+### Start the database
+
+Start the db:
+```bash
+docker run -d --rm --name iotdb -p 6667:6667 -p 9003:9003 apache/iotdb:latest
+```
+Connect to it via cli:
+```bash
+docker exec -it iotdb ./start-cli.sh -h 127.0.0.1 -p 6667 -u root -pw root
+```
+Create a database
+```bash
+create database root.Vehicles
+```
+Create desired timeseries
+
+```bash
+create timeseries root.Vehicles.Vehicle_TraveledDistance WITH DATATYPE=FLOAT, ENCODING=RLE
+create timeseries root.Vehicles.Vehicle_Speed WITH DATATYPE=FLOAT, ENCODING=RLE
+```
+
+### Start router
+
+Build the router image
+```bash
+docker build -t router .
+```
+Run the router
+```bash
+# Docker
+docker run --rm --name router -p 8080:8080 -e HANDLER_TYPE=iotdb -e IOTDB_HOST=host.docker.internal router
+# OR natively
+npm install
+HANDLER_TYPE=iotdb IOTDB_HOST=localhost npm start
+```
+
+## Backend Option 2: RealmDB
+
+
+### Prepare cloud
+
+- Ensure that in your [ATLAS cloud](https://cloud.mongodb.com/) app there is a vehicle _document_ with an `Vehicle_VehicleIdentification_VIN` in a collection named _`Vehicles`_.
+- Ensure that this document as well contains VSS data. Here you can see the data supported in this repository for a vehicle document within _Vehicles_ that should be reflected in ATLAS:
+
+```
+_id: "<SOME_STRING>" (String)
+Vehicle_Speed: <SOME_DOUBLE> (Double)
+Vehicle_TraveledDistance: "<SOME_DOUBLE>" (Double)
+```
+
+### Start router
+
+Build the router image
+```bash
+docker build -t router .
+```
+Run the router
+```bash
+# Docker
+docker run --rm --name router -p 8080:8080 -e HANDLER_TYPE=realmdb -e VERSION_REALMDB_SCHEMA=0 -e REALM_APP_ID=<YOUR-APP-ID> -e REALM_API_KEY=<YOUR-API-KEY> router
+# OR natively
+npm install
+HANDLER_TYPE=realmdb VERSION_REALMDB_SCHEMA=0 REALM_APP_ID=<YOUR-APP-ID> REALM_API_KEY=<YOUR-API-KEY> npm start
+```
+
+## Usage
+
+See [api](#api) how to interact with the router.
+
+# API
+
+Connect your own websocket client by connecting to `ws://localhost:8080`.
+
+The examples use [websocat](https://github.com/vi/websocat) and [jq](https://github.com/jqlang/jq)
+
+## Read
+
+Schema:
+```jsonc
+{
+  "type": "read",
+  "tree": "VSS",
+  "id": "123", // The VIN
+  "uuid": "testClient", // The unique client ID
+  // For reading one
+  "node": { "name": "Vehicle_Speed" },
```

Comment:
An early review observation is that in your IoTDB implementation you have chosen to replace VSS dot notation with underscores to integrate and avoid conflicts with the dot notation of the IoTDB data model. Whilst a valid approach (I considered the same when creating the Playground) it is different to the one I choose which quotes the VSS name and which is used in various places. For example, the RemotiveLabs feeder, the VISSR integration and in the online documentation. See https://covesa.github.io/cdsp/manuals/apache-iotdb/#integrating-vss-data-into-the-iotdb-data-model

So we need to have an alignment discussion to resolve this.